server/audit_math/supersimple.py

```diff
@@ -116,25 +115,46 @@ def discrepancy(
     reported: Optional[CVR],
     audited: Optional[CVR],
 ) -> Optional[Discrepancy]:
-    # Special cases: if ballot wasn't in CVR or ballot can't be found by
-    # audit board, count it as a two-vote overstatement
-    if reported is None or audited is None:
-        error = 2
-    else:
-        v_w, v_l = (
-            (reported[contest.name][winner], reported[contest.name][loser])
-            if contest.name in reported
-            # If contest wasn't on the ballot according to the CVR
-            else (0, 0)
-        )
+    def compute_error():
+        # Special cases: if ballot wasn't in CVR or ballot can't be found by
+        # audit board, count it as a two-vote overstatement
+        if reported is None or audited is None:
+            return 2
+
         a_w, a_l = (
-            (audited[contest.name][winner], audited[contest.name][loser])
+            (int(audited[contest.name][winner]), int(audited[contest.name][loser]))
             if contest.name in audited
             # If contest wasn't on the ballot according to the audit board
             else (0, 0)
         )
-        error = (v_w - a_w) - (v_l - a_l)
 
+        # Special case for ES&S overvotes/undervotes.
+        has_overvote = "o" in reported.get(contest.name, {}).values()
+        has_undervote = "u" in reported.get(contest.name, {}).values()
+        audited_votes = sum(map(int, (audited.get(contest.name, {}).values())))
+        # If the audited result correctly identified overvote/undervote, return
+        # 0 error.  Otherwise, return an error using the standard formula, but
+        # substituting in the appropriate overvotes/undervotes.
+        if has_overvote:
+            if audited_votes > 1:
+                return 0
+            else:
+                return (1 - a_w) - (1 - a_l)
```

Comment:
I think the 1 here should be a zero. An overvote isn't a valid vote for any candidate, so v_w and v_l should be zero.

Reply:
I was thinking about what would happen in a "normal" overvote case (e.g. for another CVR type), and I was thinking that the CVR would show exactly the marks made by the voter (multiple 1s), not the interpretation made of an overvote (which would be all zeros). I don't think there's anywhere that we convert the multiple 1s to all 0s, so I think this is the most faithful representation of that case. 

Luckily, whether it's 1s or 0s, I think those two parts of this equation cancel out anyway, so I don't think it matters at the end of the day:
```
(1 - a_w) - (1 - a_l)
= 1 - a_w - 1 + a_l
= 1 - 1 - a_w + a_l
= a_l - a_w
```

I just wrote it out the long way instead of simplifying so that it would be clearer to read.

Reply:
ah I see. Yeah, I think that makes sense to me then.